### PR TITLE
Improve Deploy dialog UX: loading skeletons, error states, stable layout

### PR DESCRIPTION
**Pull request by jimf**
## Summary

- Replace "No tags available" / "not deployed" flashes with skeleton
  loading states throughout the Deploy dialog (tag list, commit list,
  branch list, environment card, diff summary)
- Add inline error banners with retry buttons when tag, commit, or
  branch queries fail; silent inline message for non-blocking current
  release failure
- Fix footer shift when DiffSummary appears by reserving a fixed-height
  slot — footer stays put regardless of selection state
- Add `cursor-pointer` to all clickable list buttons (tags, commits,
  branches)
- Add Vitest test coverage for all new loading, error, and empty states

## Test plan

- [ ] Open Deploy dialog on a staging/production environment — tag list
      should show skeleton rows while loading, not "No tags available"
- [ ] Open Deploy dialog on testing — commit list should show skeleton
      rows while loading
- [ ] Select a tag/commit — footer buttons should not shift when the
      diff summary appears
- [ ] Simulate API failure (e.g. devtools offline) — tag and commit
      lists should show error banner with working Retry button
- [ ] Hover over tag/commit/branch rows — cursor should be a pointer
- [ ] Run `npm test` — 17 new tests should pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/components/deploy/DeployTab.test.tsx [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+  CurrentReleaseEnvironment,
+  DeploymentCommit,
+  DeploymentCompareResult,
+  DeploymentRef,
+  Environment,
+} from '@/types'
+
+import { DeployTab } from './DeployTab'
+
+// fallow-ignore-next-line unresolved-import
```
[the remark this answers is not in the document]
Comment (jimf) [reply src/components/deploy/DeployTab.test.tsx:17]: No change needed. You can reply to the comment explaining that fallow-ignore-next-line is a fallow-specific directive (not eslint), matching the pattern already established in DeploymentRunWatcher.test.tsx.
